documentation/man/features/kw-signature.rst

```diff
@@ -0,0 +1,112 @@
+command will add **Reviewed-by** and **Signed-off-by**, in this order,
+to the commit's trailers pointed by `HEAD` using `user.name` and
+`user.email` from git config::
+
+  $ kw signature -r -s
+
+Changing the order of the options affects the order the lines are written.
+
+Another important aspect of this command is that, since it uses `git rebase` as
+its backend, if we run it using a SHA like `kw signature -s HEAD~3` then
+only commits refered by `HEAD~2`, `HEAD~1` and `HEAD` will be affected. This is
+similar to what happens if you execute `git rebase --interactive HEAD~3`.
+
+OPTIONS
+=======
+-s, \--add-signed-off-by:
+  Adds a **Signed-off-by** trailer line to either commits or patchsets.
+  By default it uses `user.name` and `user.email` from git config to
+  write these lines if no argument is given to this option.
+  
+-r, \--add-reviewed-by:
+  Adds a **Reviewed-by** trailer line to either commits or patchsets.
+  By default it uses `user.name` and `user.email` from git config to
+  write these lines if no argument is given to this option.
+
+-a, \--add-acked-by:
+  Adds a **Acked-by** trailer line to either commits or patchsets.
+  By default it uses `user.name` and `user.email` from git config to
+  write these lines if no argument is given to this option.
+
+-t, \--add-tested-by:
+  Adds a **Tested-by** trailer line to either commits or patchsets.
+  By default it uses `user.name` and `user.email` from git config to
+  write these lines if no argument is given to this option.
+
+-C, \--add-co-developed-by:
+  Adds a **Co-developed-by** trailer line to either commits or patchsets.
+  By default it uses `user.name` and `user.email` from git config to
+  write these lines if no argument is given to this option.
+
+-R, \--add-reported-by:
+  Add a **Reported-by** trailer line to either commits or patchsets.
+  By default it uses `user.name` and `user.email` from git config to
+  write these lines if no argument is given to this option.
+
+-f, \--add-fixes:
+  Adds a **Fixes** trailer line to either commits or patchsets.
+  It requires an argument, which must be a valid commit reference, to
+  define the fixed commit's hash that follows this tag.
+
+\--verbose:
+  Displays commands executed under the hood.
+
+EXAMPLES
+========
+Adding a **Reviewed-by** line to commit pointed by ``HEAD``::
+
+  $ kw signature --add-reviewed-by='Reviewer Name <reviewer@mail.org>'
+
+Adding an **Acked-by** line starting from ``HEAD~2`` until ``HEAD``::
+
+  $ kw signature --acked-by='Some Name <example@mail.com>' HEAD~3
+
+Adding **Fixes** line to commit pointed by ``HEAD``, which fixes another
+previous commit ``90a7ba23340d``::
+
+  $ kw signature --add-fixes='90a7ba23340d'
+
+All the above options can be used to perform operations over `.patch` files as well::
+
+  $ kw signature -r'Reviewer Name <reviewer@mail.org>' example.patch
+
+  $ kw signature -a'Some Name <example@mail.org>' example.patch
+
+  $ kw signature -f'90a7ba23340d' example.patch
```

Comment:
btw, when I tried -f in a specific branch, I got this error:
![Screenshot from 2024-07-27 10-54-58](https://github.com/user-attachments/assets/ad724175-a1fd-4408-9cb4-79871b7666f8)

Reply:
I don't know how to reproduce this error, but if no warning nor errors were reported by the feature itself before reaching this sequence of git errors then I'm almost sure the command assembled in `write_all_trailers` is getting messed up somehow.

Does using `--verbose` prints anything unusual/weird? Or nothing happens at all? By looking at the command used it should print something like:
```bash
git commit --quiet --amend --no-edit --trailer "Fixes: 51fc8079c75e (\"<message>\")"
```

Reply:
Ohh ok, I think I understand that now. Each error is associated to a token in the commit message. The message is being expanded by the shell and causing that to happen when `cmd_manager` evals the command. Can you please give the message that is forcing those errors. I believe I overlooked some important details while making this code...

Reply:
I used this repository:
https://gitlab.freedesktop.org/agd5f/linux
The commit SHA is: 51fc8079c75ea89500a2953d02dcd304719f6929

The commit message is:
```
commit 51fc8079c75ea89500a2953d02dcd304719f6929
Author: Asad Kamal <asad.kamal@amd.com>
Date:   Mon Jan 8 18:34:55 2024 +0800

    Revert "drm/amdgpu: Update irq disable flow during unload"
    
    Revert the commit as the original workaround:
    commit f5c7e7797060 ("drm/amdgpu: Adjust removal control
    flow for smu v13_0_2") that caused the issue needs re-evaluation
    
    This reverts commit 2e8e792e6a51e8cb7f5f96148146b6525dbb9cef.
    
    Signed-off-by: Asad Kamal <asad.kamal@amd.com>
    Reviewed-by: Lijo Lazar <lijo.lazar@amd.com>
```